src/EnergyPlus/InternalHeatGains.cc

```diff
@@ -264,204 +247,52 @@ namespace InternalHeatGains {
-            if (Item1 > 0) {
-                state.dataHeatBal->PeopleObjects(Item).StartPtr = state.dataHeatBal->TotPeople + 1;
-                ++state.dataHeatBal->TotPeople;
-                state.dataHeatBal->PeopleObjects(Item).NumOfZones = 1;
-                state.dataHeatBal->PeopleObjects(Item).ZoneListActive = false;
-                state.dataHeatBal->PeopleObjects(Item).ZoneOrZoneListPtr = Item1;
-            } else if (ZLItem > 0) {
-                state.dataHeatBal->PeopleObjects(Item).StartPtr = state.dataHeatBal->TotPeople + 1;
-                state.dataHeatBal->TotPeople += state.dataHeatBal->ZoneList(ZLItem).NumOfZones;
-                state.dataHeatBal->PeopleObjects(Item).NumOfZones = state.dataHeatBal->ZoneList(ZLItem).NumOfZones;
-                state.dataHeatBal->PeopleObjects(Item).ZoneListActive = true;
-                state.dataHeatBal->PeopleObjects(Item).ZoneOrZoneListPtr = ZLItem;
-            } else {
-                ShowSevereError(state,
-                                CurrentModuleObject + "=\"" + AlphaName(1) + "\" invalid " + state.dataIPShortCut->cAlphaFieldNames(2) + "=\"" +
-                                    AlphaName(2) + "\" not found.");
-                ErrorsFound = true;
-                errFlag = true;
-            }
-        }
+        const std::string peopleModuleObject = "People";
+        const std::string lightsModuleObject = "Lights";
+        const std::string elecEqModuleObject = "ElectricEquipment";
+        const std::string gasEqModuleObject = "GasEquipment";
+        const std::string hwEqModuleObject = "HotWaterEquipment";
+        const std::string stmEqModuleObject = "SteamEquipment";
+        const std::string othEqModuleObject = "OtherEquipment";
+        const std::string itEqModuleObject = "ElectricEquipment:ITE:AirCooled";
+        const std::string bbModuleObject = "ZoneBaseboard:OutdoorTemperatureControlled";
+        const std::string contamSSModuleObject = "ZoneContaminantSourceAndSink:CarbonDioxide";
 
-        if (errFlag) {
-            ShowSevereError(state, std::string{RoutineName} + "Errors with invalid names in " + CurrentModuleObject + " objects.");
-            ShowContinueError(state, "...These will not be read in.  Other errors may occur.");
-            state.dataHeatBal->TotPeople = 0;
-        }
+        auto &IHGNumbers = state.dataIPShortCut->rNumericArgs;
+        auto &AlphaName = state.dataIPShortCut->cAlphaArgs;
 
-        state.dataHeatBal->People.allocate(state.dataHeatBal->TotPeople);
+        // PEOPLE: Includes both information related to the heat balance and thermal comfort
+        setupIHGZonesAndSpaces(state,
+                               peopleModuleObject,
+                               state.dataHeatBal->PeopleObjects,
+                               state.dataHeatBal->NumPeopleStatements,
+                               state.dataHeatBal->TotPeople,
+                               ErrorsFound);
```

Comment:
New function to interpret the "Zone or ZoneList or Space or SpaceList Name" field and expand the input object as needed to all applicable Spaces.